### DISCUSSION TITLE
Support header / subheader outlining

### DISCUSSION THREAD

**Discussion by classicrob** [vslinko/obsidian-outliner#53] [Ideas]
**Is your feature request related to a problem? Please describe.**
In Obsidian, a more meaningful way to outline is with headers and subheaders. All content under a header until another header at the same or higher level is considered nested

**Describe the solution you'd like**
I would like to use the same sorts of keyboard shortcuts to manipulate headers and their nested content as "items and subitems"

**Describe alternatives you've considered**
Obsidian doesn't really have a way to do it!

**Additional context**

Let's say I decide that Chapter 2 should actually be Chapter 1. Right now, the best way to do it is just cut and paste. "Swap line up" doesn't exactly work.

![image](https://user-images.githubusercontent.com/46608303/112648186-df2eba80-8e0e-11eb-928e-08b449b539dc.png)

**Comment by vslinko** [comment]
@classicrob Thank you for your request! I can understand the use case but I think it's not exactly related to Workflowy-like experience that I want to implement.

I think it's better to create another plugin like "Move heading with it's content through the document". What do you think?

**Comment by danieltomasz** [comment]
Yes, this request describes really nice workflow that is used by many users but I think as another user that it is more suited for another plugin,  "Microsoft Word" or "OneNote" style plugin rather than extending the existing one. Here is a discussion on the Obsidian forum specifying more the possible plugin
https://forum.obsidian.md/t/navigate-move-indent-and-unindent-by-heading/4202

Travestating another  plugin developer  [laimcain](https://github.com/liamcain/obsidian-calendar-plugin#weekly-notes-have-a-new-home):
>[One Thing Well philosophy](https://en.wikipedia.org/wiki/Unix_philosophy). Plugins should be modular. Some users might want to have a Workflowy-like experience and have no use for headers and subheaders. And vice versa.

**Reply by rapatel0** [reply]
I actually disagree. 

It would really be nice to migrate bullets between sections as well. you need the outliner plugin to do this or to build a interface between the proposed external plugin. 

```
# A 
- item 1 
- item 2
paragraph of text 1
# B 
paragraph of text 2 
- item 3 
   - item 3.1
```
to 

```
# A 
- item 1 
- item 2
paragraph of text 1
# B 
- item 3 (moved)
   - item 3.1
paragraph of text 2 

```
or

```
# A 
- item 1 
- item 2
- item 3 (moved again)
   - item 3.1
paragraph of text 1
# B 
paragraph of text 2 
```

Sections and bullets can naturally just be treated as higher levels of nesting as compared to bullets. a paragraph of text can just be treated kinda like a bullet.

**Comment by classicrob** [comment]
> @classicrob Thank you for your request! I can understand the use case but I think it's not exactly related to Workflowy-like experience that I want to implement.
> 
> I think it's better to create another plugin like "Move heading with it's content through the document". What do you think?

A separate plugin would be fine! Would you make it, or should I post a plugin request? I view outlining with headings and subheadings as very similar in spirit to bullet point oriented outlining (at least they are treated this way in org-mode), though it seems we differ in opinion here.

**Comment by vslinko** [comment]
I could make such plugin, but I can't promise.
As I [mentioned in Discord](https://discordapp.com/channels/686053708261228577/707816848615407697/825299153059577876), I think https://github.com/manogna4/obsidian-koncham-nav-note is better place for such functionality.

**Comment by manogna4** [comment]
> > @classicrob Thank you for your request! I can understand the use case but I think it's not exactly related to Workflowy-like experience that I want to implement.
> > I think it's better to create another plugin like "Move heading with it's content through the document". What do you think?
> 
> A separate plugin would be fine! Would you make it, or should I post a plugin request? I view outlining with headings and subheadings as very similar in spirit to bullet point oriented outlining (at least they are treated this way in org-mode), though it seems we differ in opinion here.

as @vslinko has mentioned, this is definitely in my plans for [koncham-nav-note](https://github.com/manogna4/obsidian-koncham-nav-note). I can't promise when it'll happen though.

I use headers in a similar manner to you in some notes, so I've felt the need. It's not a pressing need though, so I can't promise when I'll do that.

**Comment by vslinko** [comment]
I've converted the issue into discussions because it's very deep in my backlog and because it's possible to vote in discussions.

**Comment by vslinko** [comment]
I made the new Obsidian Zoom plugin with headings support! Please check it out https://github.com/vslinko/obsidian-zoom

It's just zooming, not moving, but it also could be helpful for people it this thread.

**Comment by KajKandler** [comment]
> I actually disagree. 
> 
> It would really be nice to migrate bullets between sections as well. you need the outliner plugin to do this or to build a interface between the proposed external plugin. 
> 
> Sections and bullets can naturally just be treated as higher levels of nesting as compared to bullets. a paragraph of text can just be treated kinda like a bullet.

_Originally posted by @rapatel0 in https://github.com/vslinko/obsidian-outliner/discussions/53#discussioncomment-1139396_

I'd like to support this thought of rapatel0. As in the MS Word world, an Outline of an article is something that would benefit note taking a lot.

With a mix of Heading and bullet point outlining, one could seamlessly
- take bullet list, 
- order it, set nesting levels
- then with the help of a keyboard shortcut change a bullet and all it's higher level entries to a heading. 
  - That would leave the bullet points below either as writing prompts or as actual bullet lists

Thinking about the UX, it might be best to have a third mode (display, edit, outline). The new outline mode would:
- Hide all paragraphs
- Leave only the headings and bullets as a seamless hierarchy to indent/outdent 
  - move all paragraphs between headings with it, once you switch back to edit or display mode.

Not sure how to deal with paragraph, list, paragraph sections.

P.S.: Thanks for a great plugin !

**Comment by johannesCmayer** [comment]
I would like to see this extension implement. If one would write another extension would you not run into the issue that you need different keybindings? But the best solution is to have the same keybinding for both of these things and make the functionality context-dependent. I would like to indent a heading and all of the subheadings, unindent them, and move the content around. I was using emacs org-mode before and this is one of the major issues I have with obsidian right now.

I just want the exact same functionality that outliner already has, but that it also works on headings.

I can use the heading shifter extension to shift the heading left or right, but it does not modify the child headings, and it does not allow you to hove the heading.
